pyfarm/jobtypes/core/jobtype.py

```diff
@@ -890,6 +890,96 @@ def set_task_progress(self, task, progress):
                     "%r." % (task["id"], error))
                 raise
 
+    @inlineCallbacks
+    def set_task_started_now(self, task):
+        """
+        Sets the time_started of the given task to the current time on the
+        master.
+        This method is useful for batched tasks, where the actual work on a
+        single task may start much later than the work the assignment as a
+        whole.
+
+        :param dict task:
+            The dictionary containing the task we're changing the start time
+            for.
+        """
+        if not isinstance(task, dict):
+            raise TypeError(
+                "Expected a dictionary for `task`, cannot set start time")
+
+        if "id" not in task or not isinstance(task["id"], INTEGER_TYPES):
+            raise TypeError(
+                "Expected to find 'id' in `task` or for `task['id']` to "
+                "be an integer.")
+
+        url = "%s/jobs/%s/tasks/%s" % (
```

Comment:
We have been using the % operator for string formatting just about everywhere in the code until now. Do you think we should switch over to string.format() in general?